src/routes.jsx

```diff
@@ -0,0 +1,319 @@
+import { matchPath, Outlet } from 'react-router-dom';
+import { PageWrap } from '@edx/frontend-platform/react';
+
+import RouteErrorBoundary from './components/app/routes/RouteErrorBoundary';
+import Root from './components/app/Root';
+import Layout from './components/app/Layout';
+import { makeRootLoader } from './components/app/routes/loaders';
+import NotFoundPage from './components/NotFoundPage';
+
+function getRouteLoader(routeLoaderFn, queryClient) {
+  if (!queryClient) {
+    return null;
+  }
+  return routeLoaderFn(queryClient);
+}
```

Comment:
[inform] When `getRoutes` is used from `env.config.js`, there is not yet access to a `queryClient`. Instead, I've made `getRoutes` conditional such that route loaders that depend on the `queryClient` are only included in the route definitions if a `queryClient` is present.